src/PKSim.Core/Services/IndividualExtractor.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using PKSim.Assets;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+
+namespace PKSim.Core.Services
+{
+   public interface IIndividualExtractor
+   {
+      void ExtractIndividualsFrom(Population population, params int[] individualIds);
+      void ExtractIndividualsFrom(Population population, IEnumerable<int> individualIds);
+   }
+
+   public class IndividualExtractor : IIndividualExtractor
+   {
+      private readonly IExecutionContext _executionContext;
+      private readonly IEntityPathResolver _entityPathResolver;
+      private readonly IIndividualTask _individualTask;
+      private readonly IContainerTask _containerTask;
+      private readonly IBuildingBlockRepository _buildingBlockRepository;
+
+      public IndividualExtractor(IExecutionContext executionContext, IEntityPathResolver entityPathResolver, IIndividualTask individualTask, IContainerTask containerTask, IBuildingBlockRepository buildingBlockRepository)
+      {
+         _executionContext = executionContext;
+         _entityPathResolver = entityPathResolver;
+         _individualTask = individualTask;
+         _containerTask = containerTask;
+         _buildingBlockRepository = buildingBlockRepository;
+      }
+
+      public void ExtractIndividualsFrom(Population population, params int[] individualIds)
+      {
+         ExtractIndividualsFrom(population, individualIds.ToList());
+      }
+
+      public void ExtractIndividualsFrom(Population population, IEnumerable<int> individualIds)
+      {
+         if (population.IsAnImplementationOf<MoBiPopulation>())
+            throw new PKSimException(PKSimConstants.Error.CannotExtractIndividualFrom(_executionContext.TypeFor(population)));
+
+         var allPopulationParameters = population.AllVectorialParameters(_entityPathResolver).ToList();
+
+         var allDistributedParameters = allPopulationParameters.Where(x => x.Formula.IsDistributed()).ToList();
+         var allConstantParameters = allPopulationParameters.Where(x => x.Formula.IsConstant()).ToList();
+         var allRemainingParameters = allPopulationParameters.Except(allConstantParameters).Except(allDistributedParameters);
+
+         //Update distributed parameters first, then constant and last all others. This ensures that formula parameters in the extracted individuals are not updated with a constant value
+         var allParametersOrderedByFormulaType = allDistributedParameters.Union(allConstantParameters).Union(allRemainingParameters).ToList();
```

Comment:
This is the idea here
Update FOrmula parameter last, so that Formula won't be overwritten if the parameter has not changed!